data_model/src/lib.rs

```diff
@@ -495,11 +492,22 @@ model! {
         Role(Box<role::Role>),
         /// [`PermissionTokenDefinition`](`permission::token::PermissionTokenDefinition`) variant.
         PermissionTokenDefinition(Box<permission::token::PermissionTokenDefinition>),
-        /// [`Validator`](`permission::Validator`) variant.
-        Validator(Box<permission::Validator>),
         /// [`Parameter`](`parameter::Parameter`) variant.
         Parameter(Box<parameter::Parameter>),
     }
+
+    /// Sized container for all possible upgradable entities.
+    #[derive(Debug, Display, Clone, PartialEq, Eq, Hash, FromVariant, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    // NOTE: Single variant enums have representation of ()
+    // Make it #[ffi_type] if more variants are added
+    #[ffi_type(opaque)]
```

Comment:
```suggestion
    // SAFETY: `UpgradableBox` has no trap representations in `permission::Validator`
    #[ffi_type(unsafe {robust})]
```

I think this should work? Rather than forcing it to be opaque